### PR TITLE
add tslint block

### PR DESCRIPTION
**Pull request by jvanbruegge**
for all typescript linting needs :)

### CODE REVIEW

**Comment by andywer** [comment]
Awesome! :)

I am on a train right now and writing from my iPhone, but if I should forget to write a main contributors list please tell me. You are definitely on it 😉

**Comment by andywer** [comment]
Feel free to add the block to the list in the main README as well :)

**Comment by andywer** [comment]
I applied two minor changes. Added an npm version badge (which will work as soon as it's published) and I replaced the `require('webpack')` by using `context.webpack`.

**Comment by SuperManitu** [comment]
Is the options plugin also working for webpack 1? Maybe I should add an e2e test

**Comment by andywer** [comment]
I was going to write this just now 😉 Could you please use the `tslint` block in the TypeScript e2e tests?

And of course, the `LoaderOptionsPlugin` will probably work with webpack 2 only... We might just copy the approach used in the [postcss block](https://github.com/andywer/webpack-blocks/blob/master/packages/postcss/index.js#L46) for now.

But that reminds me of something... I wanted to add the webpack version to the `context`. Need to open an issue for that as well.

**Comment by SuperManitu** [comment]
if we have the version we dont need to split, becaude then the object is dumped in the webpack config

**Comment by andywer** [comment]
But for webpack 1.x we will need to add the tslint config directly to the webpack config object, for webpack 2 we need to use the `LoaderOptionsPlugin`. So we need the version to distinguish which way to go, but it must be able to do both, right?

**Comment by SuperManitu** [comment]
the block yes

**Comment by SuperManitu** [comment]
I can do this when #96 is merged

**Comment by andywer** [comment]
Sure, that would be the easiest solution :)

We just have to add a peer dependency `'@webpack-blocks/core': '>= 0.4.0'` to the tslint block then.

**Comment by SuperManitu** [comment]
I rebased on #96, but the tslint-loader does not complain even if it should (mixed semicolon style in app.ts). Works in both webpack version now

**Comment by SuperManitu** [comment]
The block is now completely working, you can test it by adding a semicolon in the app.ts of the end2end test. It will break the build

**Comment by jvanbruegge** [comment]
@andywer this should be finished now, can you take another look?

**Comment by andywer** [comment]
I think this feature branch is quite broken... There are still some remains of code that is not part of the `tslint` block, but not part of the `master` branch either.

Have a look at the changes of `webpack/index.js` and `webpack2/index.js`.
You merged some other feature branch into this one before, I think that in combination with the latest rebasing broke it. Sorry 🙈

**Comment by jvanbruegge** [comment]
You are right, one wrong commit sneaked in. I deleted it.

File: packages/tslint/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+function tslint (options) {
+  options = options || {}
+
+  const loader = context => ({
+    test: context.fileType('application/x-typescript'),
+    loaders: [ 'tslint-loader' ]
+  })
+
+  const module = context => (context.webpackVersion.major === 1 ? {
```
Comment (andywer) [inline packages/tslint/index.js]: `standard` usually requires you to put a single parameter of an arrow function into parenthesis. Maybe the linting of this code does not work? 🤔

File: packages/tslint/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    preLoaders: [ loader(context) ]
+  } : {
+    loaders: [ Object.assign({}, loader(context), {
```
Comment (andywer) [inline packages/tslint/index.js]: Might be easier readable as `loaders: [ loader(context, { enforce: 'pre' }) ]`. What do you think?

Comment (andywer) [inline packages/tslint/index.js]: (2nd param for `loader()`, will be merged into the loader object using `Object.assign()`)

File: packages/tslint/index.js (hunk 1 above)
Comment (jvanbruegge) [inline packages/tslint/index.js]: I think I remember putting a semicolon there, and it was cought. Can check later

**Comment by jvanbruegge** [comment]
I tested it, standard is indeed run on the index.js file. Adding a semicolon throws an error.
I also updated the loader function, indeed it makes it a lot more readable

**Comment by andywer** [comment]
Cool 👍

And I just recognized they relaxed the arrow-function-must-have-parenthesis rule in `standard` 😉 (feross/standard#414)

**Comment by andywer** [comment]
Just changed a line in favor of more consistent code style. We should stick to one style (single-param arrow functions have parenthesis or they don't) at least within the same file 😉

**Comment by jvanbruegge** [comment]
Ok

**Comment by andywer** [comment]
Off-topic: Since you live in Munich... Do you know https://twitter.com/munichjs? :)

**Comment by jvanbruegge** [comment]
No, but it looks very interesting, especially this meetup at the google office

**Comment by andywer** [comment]
Yeah, the people attending it seemed quite interesting as well. Just stumbled across and thought you might have already been there.

**Comment by andywer** [comment]
Published!

**Comment by jvanbruegge** [comment]
yay 🎉 🎉 🎉

**Pull request merged**